pkg/framework/aries/default.go

```diff
@@ -167,6 +181,24 @@ func setAdditionalDefaultOpts(frameworkOpts *Aries) error {
 	return nil
 }
 
+func assignVerifiableStoreIfNeeded(aries *Aries, storeProvider storage.Provider) error {
+	if aries.verifiableStore != nil {
+		return nil
+	}
+
+	provider, err := context.New(context.WithStorageProvider(storeProvider))
+	if err != nil {
+		return fmt.Errorf("verifiable store initialization failed : %w", err)
+	}
+
+	aries.verifiableStore, err = verifiable.New(provider)
```

Comment:
I don't like that idea. You are creating a context that will actually satisfy everything. But actually this context contains only the storage. It will be hard to debug if someone decides to use one more dependency in the verifiable store. The code will be compiled successfully instead of point out the error. It makes code more like dynamically typed instead of statically typed.

The simplest way to fix it. Is to receive `storage.Provider` in verifiable store.
`func New(ctx storage.Provider) (*StoreImpl, error)`

@troyronda @llorllale ^

Reply:
I agree that this is not the best idea but this approach already used in project. You can check `function func createVDRI(frameworkOpts *Aries) error` in framework.go. And if you add new dependency tests will fail.

Reply:
Tests might not fail. if you have BDD tests that cover that logic then yeah. Even in that case, you need to find out why they failed. It will take some time. This time can be saved.

Reply:
This approach used a lot in this project. I can't fix this problem in this PR with 500 lines limit cause the provider for command which initialize verifiable Store has the same issue.

Reply:
@soluchok I think you're arguing for having a `context.Provider` -> `verifiable.provider` adapter.

I generally agree, but like @AlexeyBelezeko says, the approach taken in this PR is used by the framework everywhere, and I would advise to raise this point in a separate issue.